src/dioptra/restapi/queue/controller.py

```diff
@@ -53,12 +54,36 @@ def __init__(self, *args, queue_service: QueueService, **kwargs) -> None:
 
     @login_required
     @responds(schema=QueueSchema(many=True), api=api)
-    def get(self) -> list[Queue]:
-        """Gets a list of all active queues."""
+    def get(self) -> Response:
+        """Gets a page or list of all active queues."""
         log: BoundLogger = LOGGER.new(
             request_id=str(uuid.uuid4()), resource="queue", request_type="GET"
         )  # noqa: F841
         log.info("Request received")
+
+        index = request.args.get("index", -1, type=int)
+        page_length = request.args.get("page_length", -1, type=int)
+
+        if -1 in (index, page_length):
+            return self._queue_service.get_all_unlocked(log=log)  # type: ignore
+
+        data = self._queue_service.get_page(
+            index=index, page_length=page_length, log=log
+        )
+
+        is_complete = True if Queue.query.count() <= index + page_length else False
+
+        return jsonify(
```

Comment:
The returned object should probably be a schema defined in `queue/schema.py`. However I'm not sure if marshmallow will allow that, and sending back the json response might be best. I'll leave it up to @jkglasbrenner.